fiftyone/core/collections.py

```diff
@@ -9252,14 +9252,23 @@ def create_index(self, field_or_spec, unique=False, **kwargs):
 
         return name
 
-    def drop_index(self, field_or_name):
+    def drop_index(self, field_or_name, safe_mode=True):
         """Drops the index for the given field or name.
 
         Args:
             field_or_name: a field name, ``embedded.field.name``, or compound
                 index name. Use :meth:`list_indexes` to see the available
                 indexes
+            safe_mode (True): whether to use safe mode when dropping the index
         """
+        if safe_mode:
+            agreement = None
+            while agreement not in {"y", "n", "yes", "no"}:
+                agreement = input("Are you sure you want to drop the index? (y/n)?").lower()
```

Comment:
We don't have a precedent of blocking input in the SDK, I believe. I like this, but I think it will break existing automation/scripts 🤔 

We could add a `database_safe_mode` as an opt-in setting to [FiftyOneConfig](https://github.com/voxel51/fiftyone/blob/991c1d0cead89eff78a1a11467293bd886b976a1/fiftyone/core/config.py#L59) that hard errors if unwanted index commands are attempted.

Thoughts? @swheaton @minhtuev @brimoor

Reply:
yeah agree. I don't think people are going to be dropping indexes willy nilly anyways, do you?

Reply:
Hard to say. Also, since this is mostly Teams-specific, we could move this logic into a new API/Dataset permission

Reply:
We can add config var to avoid backward compatible issue with the automation script but I think this is a necessary step to make our practice in sync with MongoDB best practices; in Compass they prompt the index name to confirm index deletion.